pennylane/pulse/parametrized_evolution.py

```diff
@@ -172,30 +172,36 @@ def __init__(
             raise ValueError(
                 "All operators inside the parametrized hamiltonian must have a matrix defined."
             )
+        self._has_matrix = params is not None and t is not None
         self.H = H
-        self.params = params
         self.odeint_kwargs = odeint_kwargs
         if t is None:
             self.t = None
         else:
             self.t = jnp.array([0, t] if qml.math.ndim(t) == 0 else t, dtype=float)
-        super().__init__(wires=H.wires, do_queue=do_queue, id=id)
+        params = [] if params is None else params
+        super().__init__(*params, wires=H.wires, do_queue=do_queue, id=id)
 
     def __call__(self, params, t, **odeint_kwargs):
-        self.params = params
-        self.t = jnp.array([0, t] if qml.math.ndim(t) == 0 else t, dtype=float)
-        if odeint_kwargs:
-            self.odeint_kwargs.update(odeint_kwargs)
-        return self
+        # Need to cast all elements inside params to `jnp.arrays` to make sure they are not casted
+        # to `np.arrays` inside `Operator.__init__`
+        params = [jnp.array(p) for p in params]
+        odeint_kwargs = dict(list(self.odeint_kwargs.items()) + list(odeint_kwargs.items()))
```

Comment:
Isn't there a simpler way to merge two dicts? Does
```suggestion
        odeint_kwargs = self.odeint_kwargs | odeint_kwargs
```
work?

Reply:
The union (`|`) operator was added in python3.9 (thus we cannot use it because we need to support python3.8).

I just realised there is a simpler way of updating it:
```pycon
>>> a = {0:1, 2:3}
>>> b = {0:2, 3:4}
>>> {**a, **b}
{0: 2, 2: 3, 3: 4}
```

Reply:
Nice! Ah, right, `|` is still new to us :D :)